horace_core/sqw/PixelData/@PixelDataFileBacked/mask.m

```diff
@@ -52,29 +52,40 @@
     obj = obj.get_new_handle();
 end
 
-mask_array = obj.logical_to_normal_index_(mask_array);
-obj.num_pixels_ = numel(mask_array);
+obj.num_pixels_ = numel(keep_array);
 
 mem_chunk_size = obj.default_page_size;
 obj.data_range = obj.EMPTY_RANGE;
 
 curr = 1;
-for i = 1:mem_chunk_size:obj.num_pixels
-    block_size = min(obj.num_pixels - i + 1, mem_chunk_size);
-    data = obj.get_fields('all', mask_array(i:i+block_size));
+page = 1;
+npix = obj.num_pixels;
+for i = 1:mem_chunk_size:npix
+    obj.page_num = page;
+    block_size = min(mem_chunk_size,npix-i+1);
+    if islogical(keep_array)
+        page_keep = keep_array(i:i+block_size-1);
+    else
+        selection = keep_array >=i & keep_array <i+block_size;
+        page_keep = keep_array(selection);
```

Comment:
This is not a `logical array 0 or 1` then. The function is specified to work with such.

Reply:
This also would mean that filebacked are masked differently to memory backed

Reply:
The description have been modified

Reply:
After some thinking, I've decided to prohibit non-logical indexes. There no direct request to implement them and I do not have time to write additional tests to fix the issue. 
Fixed